test/core/xds/xds_client_test.cc

```diff
@@ -2695,6 +2719,85 @@ TEST_F(XdsClientTest, FederationChannelFailureReportedToWatchers) {
   EXPECT_TRUE(stream2->Orphaned());
 }
 
+TEST_F(XdsClientTest, AdsReadWaitsForHandleRelease) {
+  InitXdsClient();
+  // Start watches for "foo1" and "foo2".
+  auto watcher1 = StartFooWatch("foo1");
+  auto watcher2 = StartFooWatch("foo2");
+  // Watchers should initially not see any resource reported.
+  EXPECT_FALSE(watcher1->HasEvent());
+  EXPECT_FALSE(watcher2->HasEvent());
+  // XdsClient should have created an ADS stream.
+  auto stream = WaitForAdsStream();
+  ASSERT_TRUE(stream != nullptr);
+  // XdsClient should have sent a subscription request on the ADS stream.
+  auto request = WaitForRequest(stream.get());
+  ASSERT_TRUE(request.has_value());
```

Comment:
Instead of just asserting that we got a request, please check the contents of the request via `CheckRequest()`.

Same thing for the next request.